src/Hosting/Server.IntegrationTesting/src/Common/DotNetCommands.cs

```diff
@@ -46,9 +34,23 @@ public static string GetDotNetHome()
         public static string GetDotNetInstallDir(RuntimeArchitecture arch)
         {
             var dotnetDir = DotNetHome;
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            // dotnet root on helix is in a different location than on local dev
+            // Helix is
+            // sdk
+            //  /x64
+            //  /x86
+            // Local dev is
+            // .dotnet
+            //  *x64 installation here*
+            //  /x86
+            // This checks if dotnet home is either in a x86 subfolder or not.
```

Comment:
Surprised there's still a case where `DotNetHome` is incorrect. Can that be avoided?

Reply:
This problem exists due to different install locations for dotnet between helix and local dev. Do you know how we can move x64 and x86 runtime for dotnet on helix?

Reply:
Also, I don't believe this can be avoided. The issue is that the test deployers may need to publish an app of a different bitness than they are currently running as. For example. if I run build.cmd -test -arch x64, there are tests that need to know where the x86 runtime. If we set DOTNET_HOME (which we don't today) to the x64 directory, I'd need to do some nasty path traversal to find the x86 folder.

Either way, none of these solutions are looking pretty. I'm going to minimize change as much as possible and make things consistent.

Reply:
> Do you know how we can move x64 and x86 runtime for dotnet on helix?

@HaoK ?

> I'm going to minimize change as much as possible and make things consistent.

Sounds good.

Reply:
The install dir is part of the runtests script, for example https://github.com/aspnet/AspNetCore/blob/8699557a383a3b4359be8898beeb9dc1c8527ac0/eng/helix/content/runtests.cmd#L25

Reply:
As long as its in the path, I don't think generally most tests care where dotnet is installed, so as long as you update the path as well, changing the install dir should be fine